hyperspy/_components/voigt.py

```diff
@@ -16,136 +16,99 @@
 # You should have received a copy of the GNU General Public License
 # along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
 
-import numpy as np
 import math
+import numpy as np
+import sympy
 
-from hyperspy.component import Component
+from hyperspy._components.expression import Expression
+from hyperspy._components.gaussian import _estimate_gaussian_parameters
+from distutils.version import LooseVersion
 
 sqrt2pi = math.sqrt(2 * math.pi)
+sigma2fwhm = 2 * math.sqrt(2 * math.log(2))
 
 
-def voigt(x, FWHM=1, gamma=1, center=0, scale=1):
-    """Voigt lineshape.
+class Voigt(Expression):
 
-    The voigt peak is the convolution of a Lorentz peak with a Gaussian peak.
+    r"""Voigt component.
 
-    The formula used to calculate this is::
+    Symmetric peak shape based on the convolution of a Lorentzian and Normal 
+    (Gaussian) distribution. In this case using an approximate formula by David 
+    (see Notes).
 
-        z(x) = (x + 1j gamma) / (sqrt(2) sigma)
-        w(z) = exp(-z**2) erfc(-1j z) / (sqrt(2 pi) sigma)
+    .. math:: 
+        z(x) &= \frac{x + i \gamma}{\sqrt{2} \sigma} \\
+        w(z) &= \frac{e^{-z^2} \text{erf}(-i z)}{\sqrt{2 \pi} \sigma} \\
+        f(x) &= A \Re\left\{ w \left[ z(x-x_0) \right] \right\}
 
-        V(x) = scale Re(w(z(x-center)))
 
-    Parameters
-    ----------
-    gamma : real
-       The half-width half-maximum of the Lorentzian
-    FWHM : real
-       The FWHM of the Gaussian
-    center : real
-       Location of the center of the peak
-    scale : real
-       Value at the highest point of the peak
+    ============== =============
+    Variable        Parameter 
+    ============== =============
+    :math:`x_0`     centre 
+    :math:`A`       area
+    :math:`\gamma`  gamma
+    :math:`\sigma`  sigma
+    ============== =============
 
-    Notes
-    -----
-    Ref: W.I.F. David, J. Appl. Cryst. (1986). 19, 63-64
 
-    adjusted to use stddev and HWHM rather than FWHM parameters
+    Parameters
+    -----------
+    centre : float
+        Location of the maximum of the peak.
+    area : float
+        Intensity below the peak.
+    lwidth : float
```

Comment:
Any reason to use `lwidth` and `gwidth` instead of `gamma` and `sigma`?

Reply:
see comment  below